### ISSUE TITLE
Deprecate old plugin target-dir structure support

### ISSUE THREAD

**Issue by brodybits** [apache/cordova-android#580]
Old plugin target-dir structure support, which needs special remapping code in `bin/templates/cordova/lib/pluginHandlers.js` and has led to so many issues, should be deprecated with a warning message for future removal.

Plugin authors should be able to specify exactly where source files should be installed. `lib-file` elements should be used for JAR, AAR, etc. Use of `lib-file` elements for Android NDK `.so` files should be properly supported (#557).

**Comment by dpogue** [comment]
I've been thinking the same thing.

In the **next major** (cordova-android@8) we should:
* Print a visible warning for each plugin that needs its paths remapped, maybe with a link to the issues page for that plugin (if it exists in plugin.xml)
  (Note: only one warning per plugin, not one warning per remapped file!)
* Print a single visible error/warning that the project is using plugins that will not be compatible with future versions of Cordova-Android

In the **next next major** (cordova-android@9) we should:
* Remove support for the path remapping
* Maybe try to detect cases where we know path remapping is needed and print a visible error that the plugin is not compatible

For the next major, we should also make sure all of our core plugins have been updated to not need remapping

**Comment by brodybits** [comment]
Plan sounds good. I neglected to mention the possible downside of breaking a number of plugins (again), as already happened in #539, #540, #547, #572, and #574. I already warned cordova-plugin-purchase in <https://github.com/j3k0/cordova-plugin-purchase/issues/766>, we should probably warn the other major ones and publish a blog post if we decide to do this.

**Comment by dpogue** [comment]
Yeah, that's also why I'm proposing a full major cycle with noisy warnings rather than just dropping it immediately in the next major. There will be a number of unmaintained plugins that break, the best we can do it give people lots of notice and tell them which ones will be affected.

**Comment by breautek** [comment]
What is the current status of this?
